hphp/runtime/base/variable-serializer.cpp

```diff
@@ -240,12 +240,15 @@ void VariableSerializer::write(double v) {
     if (!std::isinf(v) && !std::isnan(v)) {
       char *buf;
       vspprintf(&buf, 0, "%.*k", precision, v);
+      if (m_option & k_JSON_PRESERVE_ZERO_FRACTION
+          && strchr(buf, '.') == nullptr) {
+        free(buf);
+        asprintf(&buf, "%.1f", v);
```

Comment:
You're right, the comment about precision was bogus.

Rather than reallocating the buffer, maybe move the check one line down and do something like this?

```
m_buf->append(buf);
if (m_option & k_JSON_PRESERVE_ZERO_FRACTION
    && strchr(buf, '.') == nullptr) {
  m_buf->append(".0");
}
```